#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.256846913399654 0.9274722656597488 0.27170730485035366 4.559029402086489
  position -1.9964188639190958 1.236636771849613 -0.19337620402305378
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Wall {
  translation 0 0 0.5
  name "wall(1)"
  size 0.5 0.5 0.05
}
Wall {
  translation 0.5 0 0
  rotation 0 1 0 1.5708
  size 0.5 0.5 0.05
}
Robot {
  translation 0.3 0 0
  children [
    DEF Body Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.09 0.08 0.09
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.045 0.025 0
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
        PositionSensor {
          name "ps_1"
        }
      ]
      endPoint Solid {
        translation 0.045 0.025 0
        rotation 0 0 1 1.57
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.045 0.025 0
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
        PositionSensor {
          name "ps_2"
        }
      ]
      endPoint Solid {
        translation -0.045 0.025 0
        rotation 2.463031729542013e-07 -2.461071131913704e-07 0.9999999999999393 1.5700000000000607
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "my_controller_1_2_C"
}
